Geometry/HcalCommonData/src/HcalDDDSimConstants.cc

```diff
@@ -0,0 +1,791 @@
+  }
+#ifdef DebugLog
+  edm::LogInfo ("HCalGeom") << "HcalDDDSimConstants:numberOfCells " 
+			<< cellTypes.size()  << " " << num 
+			<< " for subdetector " << subdet;
+#endif
+  return num;
+}
+
+int HcalDDDSimConstants::phiNumber(int phi, int units) const {
+
+  int iphi_skip = phi;
+  if      (units==2) iphi_skip  = (phi-1)*2+1;
+  else if (units==4) iphi_skip  = (phi-1)*4-1;
+  if (iphi_skip < 0) iphi_skip += 72;
+  return iphi_skip;
+}
+
+void HcalDDDSimConstants::printTiles() const {
+ 
+  std::cout << "Tile Information for HB from " << etaMin.at(0) << " to " << etaMin.at(0) << "\n\n";
+  for (int eta=etaMin.at(0); eta<= etaMax.at(0); eta++) {
+    int dmax = 1;
+    if (depths[0][eta-1] < 17) dmax = 2;
+    for (int depth=1; depth<=dmax; depth++) 
+      printTileHB(eta, depth);
+  }
+
+  std::cout << "\nTile Information for HE from " << etaMin.at(1) << " to " << etaMin.at(1) << "\n\n";
+  for (int eta=etaMin[1]; eta<= etaMax[1]; eta++) {
+    int dmin=1, dmax=3;
+    if (eta == etaMin[1]) {
+      dmin = 3;
+    } else if (depths[0][eta-1] > 18) {
+      dmax = 1;
+    } else if (depths[1][eta-1] > 18) {
+      dmax = 2;
+    }
+    for (int depth=dmin; depth<=dmax; depth++)
+      printTileHE(eta, depth);
+  }
+}
+
+int HcalDDDSimConstants::unitPhi(int det, int etaR) const {
+
+  double dphi = (det == static_cast<int>(HcalForward)) ? hpar->phitable[etaR-etaMin[2]] : hpar->phibin[etaR-1];
+  return unitPhi(dphi);
+}
+
+int HcalDDDSimConstants::unitPhi(double dphi) const {
+
+  const double fiveDegInRad = 2*M_PI/72;
+  int units = int(dphi/fiveDegInRad+0.5);
+  return units;
+}
+
+void HcalDDDSimConstants::initialize( void ) {
+
+  nEta      = hpar->etaTable.size();
+  nR        = hpar->rTable.size();
+  nPhiF     = nR - 1;
+  etaMin    = hpar->etaMin;
+  etaMax    = hpar->etaMax;
+  etaMin[0] = 1;
+  etaMax[1] = nEta-1;
+  etaMax[2] = etaMin[2]+nR-2;
+
+  //Layer grouping
+  for (int i=0; i<nEta-1; ++i) {
+    unsigned int k = findLayer(i+1, hpar->layerGroupEtaSim); 
+    if (k < hpar->layerGroupEtaSim.size()) { 
+      layerGroup[i] = hpar->layerGroupEtaSim[k].layerGroup;
+    } else {
+      layerGroup[i] = layerGroup[i-1]; 
+    }
+#ifdef DebugLog
+    std::cout << "HcalDDDSimConstants:Read LayerGroup" << i << ":";
+    for (unsigned int k=0; k<layerGroup[i].size(); k++) 
+      std::cout << " [" << k << "] = " << layerGroup[i][k];
+    std::cout << std::endl;
+#endif
+  }
+
+  // Geometry parameters for HF
+  dlShort   = hpar->gparHF[0];
+  zVcal     = hpar->gparHF[4];
+  dzVcal    = hpar->dzVcal;
+#ifdef DebugLog
+  std::cout << "HcalDDDSimConstants: dlShort " << dlShort << " zVcal " << zVcal
+	    << " and dzVcal " << dzVcal << std::endl;
+#endif
+
+  //Transform some of the parameters
+  maxDepth = hpar->maxDepth;
+  maxDepth[0] = maxDepth[1] = 0;
+  for (int i=0; i<nEta-1; ++i) {
+    unsigned int imx = layerGroup[i].size();
+    int laymax = (imx > 0) ? layerGroup[i][imx-1] : 0;
```

Comment:
@bsunanda - implementing inlined function layerGroup( int i, int j) will help to save memory and will be more efficient - no vector copying and caching will be needed.